packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -436,8 +441,9 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
   function makeSource(path, state) {
     const location = path.node.loc;
     if (!location) {
-      // the element was generated and doesn't have location information
-      return;
+      throw path.buildCodeFrameError(
+        "invariant: `makeSource` cannot return source information for generated paths. This is likely a bug with `@babel/helper-builder-react-jsx-experimental`.",
+      );
```

Comment:
Can we return a `jsxEmptyExpression` here?
```js
t.jsxEmptyExpression();
```
So we can make sure `makeSource` always return a valid expression that can be contained in `jsxExpressionContainer`. By doing so we don't have to detect `hasSource`, either.

Reply:
Making this change causes this error:
```
TypeError: unknown: Property arguments[4] of CallExpression expected node to be of a type ["Expression","SpreadElement","JSXNamespacedName","ArgumentPlaceholder"] but instead got "JSXEmptyExpression"
```

Reply:
@Timer 
Good catch.
```js
return path.scope.buildUndefinedNode()
```
should do the trick.

I have push a commit that revises the current test layout. `exec` is expensive because it involves creating new vm and importing all the babel polyfills. Conventionally they are meant to test the runtime behaviour instead of the complied results.